libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -5151,7 +5151,11 @@ static int32_t convert_COMMAND_LONG_loc_param(float param, bool stores_location)
     }
 
     if (stores_location) {
-        return param *1e7;
+     	float convertedValue = param *1e7;
+        if (convertedValue < INT32_MIN || convertedValue > INT32_MAX) {
+            return 0;
```

Comment:
This isn't great behaviour; we don't want vehicles deciding to fly to 0,0 because someone's screwed up on the GCS side and we interpret their bad data as 0,0.  We should be rejecting the command.